src/bp/ui-shared/src/Contents/Components/Fields/Select.tsx

```diff
@@ -1,22 +1,65 @@
-import React, { FC } from 'react'
+import { FormData, FormDynamicOptions, FormField, FormOption } from 'botpress/sdk'
+import _ from 'lodash'
+import React, { FC, Fragment, useEffect, useState } from 'react'
 
+import { lang } from '../../../translations'
 import Dropdown from '../../../Dropdown'
 import style from '../style.scss'
 import { FieldProps } from '../typings'
 
 interface SelectProps extends FieldProps {
-  options: { value: string; label: string }[]
+  data: FormData
+  field: FormField
+  axios?: any
+  parent?
+  printField?
 }
 
-const Select: FC<SelectProps> = ({ onChange, options, value }) => (
-  <Dropdown
-    filterable={false}
-    className={style.formSelect}
-    items={options}
-    defaultItem={value}
-    rightIcon="chevron-down"
-    onChange={option => onChange?.(option.value)}
-  />
-)
+const Select: FC<SelectProps> = ({ onChange, printField, parent, field, data, axios }) => {
+  const [options, setOptions] = useState<FormOption[]>([])
+
+  useEffect(() => {
+    if (field.dynamicOptions) {
+      // tslint:disable-next-line: no-floating-promises
+      loadListElements()
+    } else {
+      setOptions(field.options as FormOption[])
```

Comment:
Casting shouldn't be necessary since it's already done on the "field" property. If it whines about being maybe undefined, use ! instead (which means "we know it's defined here"